File: LeetCode/Java/0167. Two Sum II - Input array is sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        int sum;
        
        while (left < right) {
            sum = nums[left] + nums[right];
            
            if (sum < target)
                left++;
            else if (sum > target)
                right--;
            else
                return new int[] {left + 1, right + 1};
        }
        
        return new int[0];
    }
}

File: LeetCode/Python/0167. Two Sum II - Input array is sorted.py
class Solution:
    def twoSum(self, arr: List[int], target: int) -> List[int]:
        left, right = 0, len(arr) - 1
        
        while left < right:
            _sum = arr[left] + arr[right]
            
            if _sum == target:
                return [left + 1, right + 1]
            elif _sum < target:
                left += 1
            elif _sum > target:
                right -= 1
